### PR TITLE
AP-4448: Replace link_to_accessible with govuk_link_to component

### PR DESCRIPTION
**Pull request by skdejong**
## What

[AP-4448 Use link helpers (part 3)](https://dsdmoj.atlassian.net/browse/AP-4448)

Replace `link_to_accessible` with `govuk_link_to` component. 

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/providers/merits_task_lists/_task_list_item.html.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <span class="app-task-list__task-name">
     <% formatted_name = t(".#{name}") %>
     <% if status != :waiting_for_dependency %>
-      <%= govuk_link_to formatted_name, url, aria_describedby: "#{name}-status", "aria-label": formatted_name %>
+      <%= govuk_link_to formatted_name, url, aria_describedby: "#{name}-status", class: "govuk-link" %>
```
Comment (jsugarman) [inline app/views/providers/merits_task_lists/_task_list_item.html.erb:5]: doesn't `class: "govuk-link"` get added to the html implicitly?

File: app/helpers/accessible_navigation_helper.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module AccessibleNavigationHelper
-  def link_to_accessible(name = nil, options = nil, html_options = {}, &)
+  def govuk_link_to(name = nil, options = nil, html_options = {}, &)
```
Comment (jsugarman) [inline app/helpers/accessible_navigation_helper.rb:2]: should we not be using the [govuk-components govuk_link_to](https://govuk-components.netlify.app/helpers/link/) helper. This looks to be implementing our own version of it

Comment (colinbruce) [inline app/helpers/accessible_navigation_helper.rb:2]: Can this not just be deleted now?

If we rename it, won't it clash with the 'official' helper?

**Review by colinbruce — APPROVED** [review]
Looks good to me :+1:

**Review by jsugarman — APPROVED** [review]
💯

**Pull request merged**